lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -25,7 +28,16 @@ function replaceAsString(
   }
   // Look for the old version number
   const searchString = `"${oldValue}"`;
-  const newString = `"${newValue}"`;
+  let newString = `"${newValue}"`;
+
+  if (patchReg.test(oldValue)) {
+    const replaceRegex = regEx(`(patch:${depName}@(npm:)?).*#`);
```

Comment:
Can you unify `patchReg` and `replaceRegex` into a single regex?

Reply:
i need the detection regex to make sure we are in "patch"
i need the replace regex, which replaces only the version part of the regex

im not sure i know a way to do merge detection+replace

Reply:
@rarkins is there a way to do :`StartsWith(regex)`? or contains(replaceRegex),
i tried removing detection, but when `depName` has wild characters like `**/\` it will fail on some tests with Invalid Regex

Reply:
I'm not sure about that, but I think the current approach needs changing because you are doing a test with regex 1, then an exec with regex 2, and then a replace with the result of the exec without checking that it's non-null.

Reply:
@rarkins Done, i had to Escape characters in depName in order to unify them.